src/pages/SudokuPage.js

```diff
@@ -0,0 +1,86 @@
+import Cell from "../components/Cell.js";
+import NumberInput from "../components/NumberInput";
+import { useEffect, useState } from "react";
+import { v4 as uuidv4 } from "uuid";
+
+export default function SudokuPage() {
+  const [numbers, setNumbers] = useState([[], [], [], [], [], [], [], [], []]);
+  const [isLoading, setIsLoading] = useState(false);
+  let selectedCell = [];
+
+  useEffect(() => {
+    let url = "https://sugoku.herokuapp.com/board?difficulty=easy";
+    setIsLoading(true);
+    fetch(url)
+      .then((resp) => resp.json())
+      .then((data) => {
+        setNumbers(data.board);
+        localStorage.setItem("currentSudoku", JSON.stringify(data.board));
+        setIsLoading(false);
+      });
+  }, []);
+
+  function renderCellRow(rowNumber) {
+    if (isLoading || !numbers) {
+      return <p>Loading...</p>;
+    } else {
+      let mask = JSON.parse(localStorage.getItem("currentSudoku"));
+      const numbArray = numbers[rowNumber].map((number, columnNumber) => {
+        let editable = true;
+        if (mask[rowNumber][columnNumber] !== 0) {
+          editable = false;
+        }
+        return (
+          <Cell
+            key={uuidv4()}
+            value={number}
+            id={[rowNumber, columnNumber]}
+            isEditable={editable}
+            onClick={handleCellClick}
+          />
+        );
+      });
+      return numbArray;
+    }
+  }
+
+  function renderNumberBtns() {
+    const numbBtns = [...Array(9)].map((item, index) => (
+      <NumberInput
+        key={uuidv4()}
+        onClick={handleNumberInputClick}
+        value={index + 1}
+      />
+    ));
+    return numbBtns;
+  }
+
+  function handleCellClick(cellID) {
+    selectedCell = cellID;
+  }
+
+  function handleNumberInputClick(value) {
+    if (selectedCell.length === 2) {
+      let newNumbArr = [...numbers];
+      newNumbArr[selectedCell[0]][selectedCell[1]] = value;
+      setNumbers(newNumbArr);
+    }
+  }
+
+  return (
+    <div className="App">
```

Comment:
I think this div should be in the App.js :)

Reply:
Oh! Another copy - past mistake😅 Thanks for pointing out:)